pkg/distsql/context/context.go

```diff
@@ -85,6 +85,9 @@ type DistSQLContext struct {
 	SessionAlias                string
 
 	ExecDetails *execdetails.SyncExecDetails
+
+	// Only one cop-reader can use lite worker. Using lite-worker in multiple readers will affect the concurrent execution of readers.
+	TryCopLiteWorker uint32
```

Comment:
I'm inclined that the `TryCopLiteWorker` is a [request](https://github.com/pingcap/tidb/blob/a3d60bcfc6c1f18c67eeb319fc70cf26befb8687/pkg/kv/kv.go#L529-L530) level option.

Like an index lookup with 1 row result set. If `TryCopLiteWorker` is a session level option, it's only enabled once in index scan, and in table lookup, the `atomic.CompareAndSwapUint32(tryCopLiteWorker, 0, 1)` will fail and execute with multi-goroutine model.

Reply:
I'm afraid that make `TryCopLiteWorker` in `request` level option may have similar problem in this [comment](https://github.com/pingcap/tidb/pull/57522#discussion_r1875515863), there is a known case and I have added [test](https://github.com/pingcap/tidb/pull/57522/commits/a3c7e530163c2572dc988dbe908b509edae45bc3#diff-68cc9873623c1d811b385e0f6786521e0a05ff90b96678508881406c6a4d5e8dR321) for it. And I'm not sure if there are other cases like this, so I prefer keep `TryCopLiteWorker ` to be statement level.

As index lookup with 1 row result set, you are right, but because of the above problems, I don't have a good way to fix this.